src/main/java/com/microsoft/azure/vmagent/AzureVMAgent.java

```diff
@@ -220,6 +228,24 @@ public AzureVMAgent(
         this.creationTime = System.currentTimeMillis();
     }
 
+    private static DescribableList<NodeProperty<?>, NodePropertyDescriptor> getNodeProperties(
+            String fqdn, AzureVMAgentTemplate template
+    ) {
+        DescribableList<NodeProperty<?>, NodePropertyDescriptor> nodeProperties = template.getNodeProperties();
+        EnvironmentVariablesNodeProperty envVarsNodeProperty = nodeProperties
+                .get(EnvironmentVariablesNodeProperty.class);
+
+        List<EnvironmentVariablesNodeProperty.Entry> newEntries = new ArrayList<>();
+        if (envVarsNodeProperty != null) {
+            nodeProperties.remove(envVarsNodeProperty);
+            newEntries.addAll(envVarsNodeProperty.getEnv());
+        }
+        newEntries.add(new EnvironmentVariablesNodeProperty.Entry("FQDN", fqdn));
```

Comment:
merges the one this plugin previously set with the ones potentially in the template